ts/IdentificationModal.tsx

```diff
@@ -0,0 +1,191 @@
+
+import Pinpad from "./components/Pinpad";
+import BaseScreenComponent from "./components/screens/BaseScreenComponent";
+import IconFont from "./components/ui/IconFont";
+import TextWithIcon from "./components/ui/TextWithIcon";
+import I18n from "./i18n";
+import {
+  identificationCancel,
+  identificationFailure,
+  identificationPinReset,
+  identificationSuccess
+} from "./store/actions/identification";
+import { ReduxProps } from "./store/actions/types";
+import { IdentificationState } from "./store/reducers/identification";
+import { GlobalState } from "./store/reducers/types";
+
+type MapStateToProps = {
+  identification: IdentificationState;
+};
+
+type Props = MapStateToProps & ReduxProps;
+
+type IdentificationByPinState = "unstarted" | "failure";
+
+type State = {
+  identificationByPinState: IdentificationByPinState;
+};
+
+const contextualHelp = {
+  title: I18n.t("pin_login.unlock_screen.help.title"),
+  body: () => I18n.t("pin_login.unlock_screen.help.content")
+};
+
+const onRequestCloseHandler = () => undefined;
+
+const renderIdentificationByPinState = (
+  identificationByPinState: IdentificationByPinState
+) => {
+  if (identificationByPinState === "failure") {
+    return (
+      <React.Fragment>
+        <View spacer={true} extralarge={true} />
+        <TextWithIcon danger={true}>
+          <IconFont name="io-close" color={"white"} />
+          <Text white={true}>{I18n.t("pin_login.pin.confirmInvalid")}</Text>
+        </TextWithIcon>
+      </React.Fragment>
+    );
+  }
+
+  return null;
+};
+
+class IdentificationModal extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      identificationByPinState: "unstarted"
+    };
+  }
+
+  public render() {
+    const { identification, dispatch } = this.props;
+
+    // If the identification state is started we need to show the modal
+    if (identification.kind === "started") {
+      const {
+        pin,
+        identificationCancelData,
+        identificationSuccessData
+      } = identification;
+
+      const { identificationByPinState } = this.state;
+
+      /**
+       * Create handlers merging default internal actions (to manage the identification state)
+       * with, if available, custom actions passed as props.
+       */
+      const onIdentificationCancelHandler = () => {
+        if (identificationCancelData) {
+          dispatch(identificationCancelData.action);
+        }
+        dispatch(identificationCancel());
+      };
+
+      const onIdentificationSuccessHandler = () => {
+        if (identificationSuccessData) {
+          dispatch(identificationSuccessData.action);
+        }
+        dispatch(identificationSuccess());
+      };
+
+      const onIdentificationFailureHandler = () => {
+        dispatch(identificationFailure());
+      };
+
+      const onPinResetHandler = () => {
+        dispatch(identificationPinReset());
+      };
+
+      return (
+        <Modal
```

Comment:
why do we need a modal? can't this be just a screen rendered on top of everything?

Reply:
A screen? You mean a react-navigation screen or something else? The purpose is to leave the AppNavigator untouched and just display something that force the user to identify.

Reply:
I mean something like `ConnectionBar` or `VersionInfoOverlay`